xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-filters/xwiki-platform-notifications-filters-default/src/main/java/org/xwiki/notifications/filters/internal/NotificationFilterPreferenceStore.java

```diff
@@ -220,6 +224,44 @@ public void deleteFilterPreference(DocumentReference user, String filterPreferen
         this.deleteFilterPreference(preference);
     }
 
+    /**
+     * Delete all to notification preferences where the given user is the owner.
+     *
+     * @param user the document reference of a user
+     * @throws NotificationException in case of error during the hibernate operations
+     * @since 14.5RC1
+     * @since 14.4.1
+     * @since 13.10.7
+     */
+    public void deleteFilterPreferences(DocumentReference user) throws NotificationException
+    {
+        XWikiContext context = this.contextProvider.get();
+
+        String oriDatabase = context.getWikiId();
+
+        context.setWikiId(context.getMainXWiki());
```

Comment:
not sure to understand: aren't the filter preferences stored in the wiki where the user is registered? So in case of a user of a subwiki, shouldn't it be stored in subwiki DB?

Reply:
Hum, looks like a bug indeed, fixing.

Reply:
Note that the decision to use the main database for sub-wikis should be based on the `eventstream.usemainstore` property.

Reply:
> Note that the decision to use the main database for sub-wikis should be based on the eventstream.usemainstore property.

I wasn't aware about this option, but when I read about it it's only talking about storage of events, it's also used for storage of filter preferences?